lib/matplotlib/figure.py

```diff
@@ -1038,6 +1038,42 @@ def fixlist(args):
         key = fixlist(args), fixitems(kwargs.items())
         return key
 
+    def add_artist(self, artist, clip=False):
+        """
+        Add any :class:`~matplotlib.artist.Artist` to the figure.
+
+        Usually artists are added to axes objects using
+        :meth:`matplotlib.axes.Axes.add_artist`, but use this method in the
+        rare cases that adding directly to the figure is necessary.
+
+        Parameters
+        ----------
+        artist : `~matplotlib.artist.Artist`
+            The artist to add to the figure. If the added artist has no
+            transform set to it previously, its transform will be set to
+            ``figure.transFigure``.
+        clip : bool, optional, default ``False``
```

Comment:
I don't understand this parameter for a Figure artist; I would not expect that anything could be drawn outside the Figure patch boundaries.  I did a little test that seemed to confirm that even with the ps backend, so there is white space around the figure, a Figure.figimage is clipped, regardless of whether get_clip_on() returns True or False.

Reply:
What if you do `savefig(bbox_inches='tight')`?  If clipping is on, it'll clip the tight bbox (I think, or maybe it will when some of my PRs on this get merged).  If clipping is off the figure will get larger...